File: app/src/main/java/com/example/final_pocproject/activities/MainActivity.java
package com.example.final_pocproject.activities;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.util.Patterns;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.example.final_pocproject.api.ApiInterface;
import com.example.final_pocproject.R;
import com.example.final_pocproject.interfaces.UserInterface;
import com.example.final_pocproject.model.User;
import com.example.final_pocproject.api.RetrofitClient;
import com.example.final_pocproject.utility.UserUtility;
import com.google.android.material.textfield.TextInputLayout;

import java.util.List;
import java.util.Objects;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity implements UserInterface {


    private Button loginButton;
    private TextInputLayout textInputEmail;
    UserUtility userUtility;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        loginButton = findViewById(R.id.button);
        textInputEmail = findViewById(R.id.email);
        userUtility = new UserUtility(this);

        loginButton.setOnClickListener(new View.OnClickListener() {
            @RequiresApi(api = Build.VERSION_CODES.KITKAT)
            @Override
            public void onClick(View v) {
                String inputEmail = Objects.requireNonNull(textInputEmail.getEditText()).getText().toString().trim();

                if (validateEmail(inputEmail)) {
                    userUtility.getUser();
                }
            }
        });

    }

    private boolean validateEmail(String inputEmail) {

        if (inputEmail.isEmpty()) {
            textInputEmail.setError("Email field can't be empty");
            return false;
        } else if (!Patterns.EMAIL_ADDRESS.matcher(inputEmail).matches()) {
            textInputEmail.setError("Please enter a valid email id");
            return false;
        } else {
            textInputEmail.setError(null);
            return true;
        }
    }

    @Override
    public void handleSuccessResponse(List<User> users) {

        if (users == null) {
            Toast.makeText(MainActivity.this, "Posts could not be loaded", Toast.LENGTH_SHORT).show();
        }

        String email = textInputEmail.getEditText().getText().toString();

        int userId = 0;
        for (User user : users) {
            if (email.equalsIgnoreCase(user.getEmail())) {
                userId = user.getId();

                break;
            }
        }

        if (userId == 0) {
            textInputEmail.setError("Email id not found");
        } else {
            Intent intent = new Intent(MainActivity.this, PostActivity.class);
            intent.putExtra("id", userId);
            startActivity(intent);
        }
    }

    @Override
    public void handleFailure(Throwable t) {
        Toast.makeText(MainActivity.this, t.getMessage(), Toast.LENGTH_SHORT).show();
    }

}

File: app/src/main/java/com/example/final_poc_project/activities/MainActivity.kt
package com.example.final_poc_project.activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import com.example.final_poc_project.*
import com.example.final_poc_project.interfaces.UserInterface
import com.example.final_poc_project.model.User
import com.example.final_poc_project.api.UserApi
import com.example.final_poc_project.sessionmanagement.SessionManager
import com.google.android.material.textfield.TextInputLayout


class MainActivity : AppCompatActivity(), UserInterface {

    lateinit var loginButton: Button
    lateinit var inputEmail: TextInputLayout
    private val emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+"
    lateinit var userApi: UserApi
    lateinit var sessionManager: SessionManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sessionManager = SessionManager(applicationContext)
        if (sessionManager.isLoggedIn()) {
            val intent: Intent = Intent(this, DashboardActivity::class.java)
            finish()
            startActivity(intent)
        }

        loginButton = findViewById(R.id.loginButton)
        inputEmail = findViewById(R.id.email)
        userApi = UserApi(this)

        loginButton.setOnClickListener(View.OnClickListener {

            val emailId: String = inputEmail.editText?.text.toString().trim()

            if (validEmail(emailId))
                userApi.getUser()
        })
    }


    private fun validEmail(emailId: String): Boolean {
        var isValid: Boolean = false
        if (emailId.isEmpty()) {
            inputEmail.error = getString(R.string.emailErrorMessage)
        } else if (emailId.matches(emailPattern.toRegex())) {
            inputEmail.error = null
            isValid = true
        } else {
            inputEmail.error = getString(R.string.invalidEmail)
        }
        return isValid
    }

    override fun handleSuccessResponse(users: List<User>) {
        val emailEntered: String = inputEmail.editText?.text.toString()
        var userId: Int = 0
        for (i: Int in users.indices)
            if (users[i].email.equals(emailEntered, true)) {
                userId = users[i].id
            }

        if (userId != 0) {
            sessionManager.createLoginSession(emailEntered, userId)
            val intent = Intent(this@MainActivity, DashboardActivity::class.java)
            finish()
            startActivity(intent)
        } else {
            inputEmail.error = getString(R.string.emailNotFound)
        }
    }

    override fun handleFailure(t: Throwable) {
        Toast.makeText(this@MainActivity, t.message, Toast.LENGTH_SHORT).show()
    }
}

